File: backend/src/main/java/de/neuefische/backend/model/Course.java
package de.neuefische.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "courses")
public class Course {
    @Id
    private String id;
    private CourseType courseType;
    private CourseLevel courseLevel;
    private String instructorId;
    private List<Participant> participants;
    boolean completed;
}

File: frontend/src/models/Course.ts
import {Participant} from "./Participant.ts";
import {CourseLevel} from "./CourseLevel.ts";
import {CourseType} from "./CourseType.ts";

export type Course={
    id: string,
    courseType: CourseType,
    courseLevel: CourseLevel,
    instructorId: string,
    participants: Participant[],
    completed: boolean,
}